### PR TITLE
nix CI

### PR DESCRIPTION
**Pull request by bgregoir**
With Cyril help's

### CODE REVIEW

**Comment by eponier** [comment]
We discussed with @CohenCyril about how to use coq-nix-toolbox to both have `nix-shell` and `ci`. This PR is a proof of concept.

@CohenCyril explained that by default there is no caching. If we want some, we would have to ask for an access for Coq's cache or create our own account elpi/coq-elpi on Cachix. We can also disable the test of reverse dependencies if we find it too costly.

**Comment by CohenCyril** [comment]
@Zimmi48 could we make room for a LPCIC cache in the coq cachix organization?

**Comment by eponier** [comment]
@gares To be fair w.r.t. Cyril's work, the effort was pretty small. It was basically a matter of following https://github.com/coq-community/coq-nix-toolbox#standalone. With Cyril's help, it was straightforward. But it may become more complex for more advanced stuff, I don't know.

**Comment by gares** [comment]
Fine with nix, but please don't remove the other workflows. One builds the doc, which I don't think nix does.
The other is probably subsumed by nix.

**Comment by gares** [comment]
I did the fix myself, if it works I'll merge

**Comment by Zimmi48** [comment]
Sure, we can add @gares to the "Coq ecosystem" Cachix organization. Please contact me by private message on Zulip.

**Comment by gares** [comment]
@CohenCyril why don't you become an admin of this repo and set up cachix?

**Comment by CohenCyril** [comment]
> @CohenCyril why don't you become an admin of this repo and set up cachix?

Sure admin me

**Comment by gares** [comment]
Apparently there is a "maintainer" role, but I could not understand if you can create secrets. Please try, if not I'll admin you.

**Comment by CohenCyril** [comment]
@gares I do not have access to secrets

**Comment by gares** [comment]
ok, now you are Admin.

File: .nix/config.nix [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  ## write one `bundles.name` attribute set per
+  ## alternative configuration
+  ## When generating GitHub Action CI, one workflow file
+  ## will be created per bundle
+  bundles.default = {
+
+    ## You can override Coq and other Coq coqPackages
+    ## through the following attribute
+    coqPackages.coq.override.version = "8.15";
+    #coqPackages.hierarchy-builder.job = false;
```
Comment (CohenCyril) [inline .nix/config.nix:45]: @gares
```suggestion
    coqPackages.hierarchy-builder.override.version = "master";
```

**Review by CohenCyril — COMMENTED** [review]
@gares to reactivate rev dep test on HB for example

**Pull request merged**